test/integration/whitespace.js

```diff
@@ -7,33 +7,43 @@ const minVersion = require('../../ranges/min-version')
 const minSatisfying = require('../../ranges/min-satisfying')
 const maxSatisfying = require('../../ranges/max-satisfying')
 
-const s = (n = 500000) => ' '.repeat(n)
+const wsMedium = ' '.repeat(125)
+const wsLarge = ' '.repeat(500000)
+const zeroLarge = '0'.repeat(500000)
 
-test('regex dos via range whitespace', (t) => {
-  // a range with this much whitespace would take a few minutes to process if
+test('range with whitespace', (t) => {
+  // a range with these extra characters would take a few minutes to process if
   // any redos susceptible regexes were used. there is a global tap timeout per
   // file set in the package.json that will error if this test takes too long.
-  const r = `1.2.3 ${s()} <1.3.0`
-
+  const r = `1.2.3 ${wsLarge} <1.3.0`
   t.equal(new Range(r).range, '1.2.3 <1.3.0')
   t.equal(validRange(r), '1.2.3 <1.3.0')
   t.equal(minVersion(r).version, '1.2.3')
   t.equal(minSatisfying(['1.2.3'], r), '1.2.3')
   t.equal(maxSatisfying(['1.2.3'], r), '1.2.3')
+  t.end()
+})
 
+test('range with 0', (t) => {
+  const r = `1.2.3 ${zeroLarge} <1.3.0`
+  t.throws(() => new Range(r).range)
+  t.equal(validRange(r), null)
+  t.throws(() => minVersion(r).version)
+  t.equal(minSatisfying(['1.2.3']), null)
+  t.equal(maxSatisfying(['1.2.3']), null)
```

Comment:
these should probably use `r` 😉

Reply:
Good catch! https://github.com/npm/node-semver/pull/574